docs/sources/k6/next/misc/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,294 @@
+
+This includes the [go.k6.io/k6/js/modules](https://pkg.go.dev/go.k6.io/k6@v0.57.0/js/modules) Go package that internal and external developers have available to build module extensions, as well as the [go.k6.io/k6/output](https://pkg.go.dev/go.k6.io/k6@v0.57.0/output) that can be used to build custom outputs for k6.
+
+It also includes public k6 Go packages that are exposed to developers to facilitate the development of these extensions. For example, the current  [go.k6.io/k6/lib.State](https://pkg.go.dev/go.k6.io/k6@v0.57.0/lib#State) package.
+
+### Output
+
+We guarantee the stability and support of metrics, logs, traces, and profiling outputs that we fully control in an end-to-end flow. This includes the JSON, CSV, cloud metrics, logs and traces outputs, and OpenTelemetry formats while excluding outputs dependent on third-party platforms and protocols.
+
+### Metrics
+
+We guarantee the stability of metric names, types, and units. Any changes to these are considered breaking and will be handled according to our [versioning strategy](#versioning-strategy). Doing so ensures that any product or artifact built on top of k6, such as dashboards, can maintain a reliable and consistent experience.
+
+## Deprecation, breaking changes, and migration policy
+
+### Breaking changes
+
+In a constantly evolving product, [breaking changes](#breaking-change) are sometimes unavoidable. They require that k6 users actively react to them because the state after the changes is not backward compatible with the previous state.
+
+Because we follow the [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html) versioning scheme, we work under the assumption breaking changes will always lead to a new major version. When the k6 team decides to introduce a breaking change, it will automatically define the need for a new major version.
+
+When k6 releases a new major version, we are expected to be explicit, clearly communicate the breaking changes, and provide instructions for migrating users.
+
+There are two types of breaking changes:
+
+1. Removal
+2. Update
+
+#### Steps to remove an API or a feature
+
+1. Deprecate a current API or a feature by following the dedicated policy.
+2. If necessary, add an alternative solution. Ideally, this would be an experimental version within a minor release within the current maintained state. Alternatively, it might be available in a Preview release.
+3. Provide documentation for the migration path.
+4. Proceed with removing the API and releasing a new major version.
+
+#### Steps to update an API or a feature
+
+1. Apply the changes, then publish them in the next major version. If the change is significant, it might be convenient to make it available in a Preview release that can support the migration path until a version is globally available.
+2. Provide documentation for the migration path.
```

Comment:
```suggestion
1. Apply the changes, then publish them in the next major version. If the change is significant, it might be convenient to make it available in a Preview release that can support the migration path until a version is globally available.
1. Provide documentation for the migration path.
```